shell/src/main/java/alluxio/shell/command/HelpCommand.java

```diff
@@ -0,0 +1,133 @@
+import alluxio.exception.AlluxioException;
+import alluxio.util.CommonUtils;
+
+import com.google.common.base.Throwables;
+import org.apache.commons.cli.CommandLine;
+import org.reflections.Reflections;
+
+import java.io.IOException;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Command for print help message for the given command.
+ * If there isn't given command, print help messages for all supported commands.
+ */
+@ThreadSafe
+public final class HelpCommand extends AbstractShellCommand {
+
+  /**
+   * @param fs the filesystem of Alluxio
+   */
+  public HelpCommand(FileSystem fs) {
+    super(fs);
+  }
+
+  @Override
+  public String getCommandName() {
+    return "help";
+  }
+
+  @Override
+  public boolean validateArgs(String... args) {
+    boolean valid = args.length >= getNumOfArgs();
+    if (!valid) {
+      System.out.println(getCommandName() + " takes at least " + getNumOfArgs() + " arguments, "
+              + " not " + args.length + "\n");
+    }
+    return valid;
+  }
+
+  private final Map<String, ShellCommand> mCommands = new HashMap<>();
+
+  @Override
+  public void run(CommandLine cl) throws AlluxioException, IOException {
+    String[] args = cl.getArgs();
+    loadCommands();
+    SortedSet<String> sortedCmds = null;
+    boolean hasUnknownCommand = false;
+    if (args.length == 0) {
+      //print help messages for all supported commands.
+      sortedCmds = new TreeSet<>(mCommands.keySet());
+    } else {
+      sortedCmds = new TreeSet<>();
+      for (String cmd : args) {
+        if (mCommands.containsKey(cmd)) {
+          sortedCmds.add(cmd);
+        } else {
+          hasUnknownCommand = true;
+          System.out.println(cmd + " is an unknown command.");
+        }
+      }
+    }
+    if (hasUnknownCommand) {
+      System.out.println();
+    }
+    printUsage(sortedCmds);
+  }
+
+  private void loadCommands() {
+    String pkgName = ShellCommand.class.getPackage().getName();
+    Reflections reflections = new Reflections(pkgName);
+    for (Class<? extends ShellCommand> cls : reflections.getSubTypesOf(ShellCommand.class)) {
+      // Only instantiate a concrete class
+      if (!Modifier.isAbstract(cls.getModifiers())) {
+        ShellCommand cmd;
+        try {
+          cmd = CommonUtils.createNewClassInstance(cls,
+              new Class[] { FileSystem.class },
+              new Object[] {mFileSystem });
+        } catch (Exception e) {
+          throw Throwables.propagate(e);
+        }
+        mCommands.put(cmd.getCommandName(), cmd);
+      }
+    }
+  }
+
+  private void printUsage(SortedSet<String> sortedCmds) {
+    if (sortedCmds == null || sortedCmds.size() == 0) {
+      return;
+    }
+    for (String cmd : sortedCmds) {
+      System.out.format("%-60s%-95s%n", "       [" + mCommands.get(cmd).getUsage() + "]   ",
```

Comment:
The origin print layout in 'AlluxioShell.java' just like that... I will remove this limitation of description.